modules/git/repo_attribute.go

```diff
@@ -205,7 +205,7 @@ func (c *CheckAttributeReader) Run() error {
 			return nil
 		},
 	})
-	if err != nil && c.ctx.Err() != nil && err.Error() != "signal: killed" {
+	if err != nil && c.ctx.Err() != err && err.Error() != "signal: killed" {
```

Comment:
It's difficult for a reader to understand what does `c.ctx.Err() != err` mean and why it should be ignored here ..... I think we need more comments for reasons and correctness.